torch/csrc/jit/passes/alias_analysis.cpp

```diff
@@ -307,15 +321,41 @@ void AliasDb::analyzeImpl(Node* node) {
       if (tryRegisteredAnalysis(node)) {
         return;
       }
-      TORCH_INTERNAL_ASSERT(!aliasAnalysisHasSpecialCaseFor(node->kind()));
   }
 
-  const auto& schema = node->schema();
-  // see [custom operator aliasing]
-  if (!node->kind().is_aten() && !node->kind().is_prim()) {
+  TORCH_INTERNAL_ASSERT(op, "We should have an op schema if we get to here");
+  const AliasAnalysisKind analysis = op->aliasAnalysisKind();
+  TORCH_INTERNAL_ASSERT(analysis != AliasAnalysisKind::INTERNAL_SPECIAL_CASE && !aliasAnalysisHasSpecialCaseFor(node->kind()), "Special cases should be handled already if we're here.");
+
+  if (node->kind().is_aten() || node->kind().is_prim()) {
+    // TODO This assert is only introduced to check that we don't break the current code base. Remove this later to allow aten:: and prim:: ops to use other alias analysis kinds.
+    TORCH_INTERNAL_ASSERT(analysis == AliasAnalysisKind::FROM_SCHEMA, "aten:: and prim:: operators should use AliasAnalysisKind::FROM_SCHEMA but ", node->kind().toDisplayString(), " doesn't.");
```

Comment:
Primitive ops are defined primarily by not being schematizable. So it's misleading somewhat to say that prim:: operators should use FROM_SCHEMA (as opposed to having a special case registered)

Reply:
There are a few prim:: ops that have schemas. I clarified the comment.